op-node/rollup/driver/state_tob_test.go

```
@@ -0,0 +1,253 @@
+// - Metrics (only mocked enough to let the method proceed)
+// - Publishing (network is set to nil so publishing won't occur)
+func TestRejectCreateBlockBadTimestamp(t *testing.T) {
+	// Create our random provider
+	rng := rand.New(rand.NewSource(rand.Int63()))
+
+	// Create our context for methods to execute under
+	ctx := context.Background()
+
+	// Create our fake L1/L2 heads and link them accordingly
+	l1HeadRef := testutils.RandomBlockRef(rng)
+	l2HeadRef := testutils.RandomL2BlockRef(rng)
+	l2l1OriginBlock := l1HeadRef
+	l2HeadRef.L1Origin = l2l1OriginBlock.ID()
+
+	// Create a rollup config
+	cfg := rollup.Config{
+		BlockTime: uint64(60),
+		Genesis: rollup.Genesis{
+			L1:     l1HeadRef.ID(),
+			L2:     l2HeadRef.ID(),
+			L2Time: 0x7000, // dummy value
+		},
+	}
+
+	// Patch our timestamp so we fail
+	l2HeadRef.Time = l2l1OriginBlock.Time - (cfg.BlockTime * 2)
+
+	// Create our outputter
+	outputProvider := TestDummyOutputImpl{willError: false}
+
+	// Create our state
+	s := Driver{
+		l1State: &L1State{
+			l1Head:  l1HeadRef,
+			log:     log.New(),
+			metrics: &metrics.Metrics{TransactionsSequencedTotal: prometheus.NewCounter(prometheus.CounterOpts{})},
+		},
+		log:              log.New(),
+		l1OriginSelector: TestDummyL1OriginSelector{retval: l1HeadRef},
+		config:           &cfg,
+		sequencer:        outputProvider,
+		derivation:       TestDummyDerivationPipeline{},
+		metrics:          &metrics.Metrics{TransactionsSequencedTotal: prometheus.NewCounter(prometheus.CounterOpts{})},
+	}
+
+	// Create a new block
+	// - L2Head's L1Origin, its timestamp should be greater than L1 genesis.
+	// - L2Head timestamp + BlockTime should be greater than or equal to the L1 Time.
+	err := s.createNewL2Block(ctx)
+
+	// Verify the L1Origin's timestamp is greater than L1 genesis in our config.
+	if l2l1OriginBlock.Number < s.config.Genesis.L1.Number {
+		assert.Nil(t, err)
+		return
```

`require.NoError(t, err, "message why we don't expect an error here")` would be more idiomatic. The `require` essentially asserts and fails the test there, instead of having to return to stop execution.